aws/dx_vif.go

```diff
@@ -0,0 +1,168 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dxVirtualInterfaceRead(id string, conn *directconnect.DirectConnect) (*directconnect.VirtualInterface, error) {
+	resp, state, err := dxVirtualInterfaceStateRefresh(conn, id)()
+	if err != nil {
+		return nil, fmt.Errorf("Error reading Direct Connect virtual interface: %s", err)
+	}
+	if state == directconnect.VirtualInterfaceStateDeleted {
+		return nil, nil
+	}
+
+	return resp.(*directconnect.VirtualInterface), nil
+}
+
+func dxVirtualInterfaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	arn := arn.ARN{
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Service:   "directconnect",
+		AccountID: meta.(*AWSClient).accountid,
+		Resource:  fmt.Sprintf("dxvif/%s", d.Id()),
+	}.String()
+	if err := setTagsDX(conn, d, arn); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func dxVirtualInterfaceDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	log.Printf("[DEBUG] Deleting Direct Connect virtual interface: %s", d.Id())
+	_, err := conn.DeleteVirtualInterface(&directconnect.DeleteVirtualInterfaceInput{
+		VirtualInterfaceId: aws.String(d.Id()),
+	})
+	if err != nil {
+		if isAWSErr(err, "DirectConnectClientException", "does not exist") {
```

Comment:
Minor nitpick: looks like the SDK has an available constant for this: `directconnect.ErrCodeClientException`

Reply:
👍